internal/expr/parser_helper_test.go

```diff
@@ -243,3 +243,65 @@ func (s parseSuite) TestParseSliceRange(c *C) {
 		}
 	}
 }
+
+func (s parseSuite) TestAdvanceToNextExpr(c *C) {
+	var p = NewParser()
+
+	tests := []struct {
+		input   string
+		stopPos []int
+	}{{
+		// advanceToNextExpr stops on the first char if it is the start of the
+		// expr.
+		input:   `col1`,
+		stopPos: []int{0},
+	}, {
+		input:   `&col`,
+		stopPos: []int{0},
+	}, {
+		input:   `$col`,
+		stopPos: []int{0},
+	}, {
+		input:   `word,&`,
+		stopPos: []int{5},
+	}, {
+		input:   ` ,&`,
+		stopPos: []int{2},
+	}, {
+		input:   ` col1`,
+		stopPos: []int{1},
+	}, {
+		input:   `/* &Person.* */`,
+		stopPos: []int{},
+	}, {
+		input:   `" &Person.*"`,
+		stopPos: []int{},
+	}, {
+		input:   ` ""`,
+		stopPos: []int{},
+	}, {
+		input:   ` /**/`,
+		stopPos: []int{},
+	}}
+
+	for _, t := range tests {
+		p.init(t.input)
+		currentStopPos := 0
+		for p.pos < len(p.input) {
+			err := p.advanceToNextExpression()
+			c.Assert(err, IsNil)
+			if p.pos >= len(p.input) {
+				break
+			}
```

Comment:
Maybe I am misunderstanding it but I think this line is somewhat bypassing the tests. If I changed the implementation so that `&col` did not stop I think the tests will still succeed. I think we need to assert that `currentStopPos == len(t.stopPos)` here.